src/components/StateContext.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { createContext, useEffect, useReducer } from 'react';
+import { Action } from '../types/Action';
+import { Todo } from '../types/Todo';
+import { FilterType } from '../types/FilterType';
+
+type StateContextType = {
+  todos: Todo[];
+  toDoTitle: string;
+  filter: FilterType;
+  edittedTitle: string;
+  focusOnTodo: boolean;
+};
+
+const reducer = (state: StateContextType, action: Action): StateContextType => {
+  switch (action.type) {
+    case 'FOCUS_ON_TODO':
+      return {
+        ...state,
+        focusOnTodo: !state.focusOnTodo,
+      };
```

Comment:
this one action is weird. I think it can be handled with easier solution

Reply:
When deleted this action, I didnt know how to focus header input. I can do this only with useState but I use reducer to avoid this.